pgmpy/models/NaiveBayesModel.py

```diff
@@ -0,0 +1,224 @@
+import itertools
+from collections import defaultdict
+import logging
+
+import networkx as nx
+import numpy as np
+import pandas as pd
+
+from pgmpy.base import DirectedGraph
+from pgmpy.factors import TabularCPD
+from pgmpy.independencies import Independencies
+from pgmpy.independencies import IndependenceAssertion
+from pgmpy.extern.six.moves import range
+from pgmpy.models import BayesianModel
+
+class NaiveBayesModel(BayesianModel):
+    """
+    Class to represent Naive Bayes.
+    Subclass of Bayesian Model.
+    Model holds directed edges from one parent node to multiple
+    children nodes only.
+
+    Parameters
+    ----------
+    data : input graph
+        Data to initialize graph.  If data=None (default) an empty
+        graph is created.  The data can be an edge list, or any
+        NetworkX graph object.
+
+    Examples
+    --------
+    Create an empty Naive Bayes Model with no nodes and no edges.
+
+    >>> from pgmpy.models import NaiveBayesModel
+    >>> G = NaiveBayesModel()
+
+    G can be grown in several ways.
+
+    **Nodes:**
+
+    Add one node at a time:
+
+    >>> G.add_node('a')
+
+    Add the nodes from any container (a list, set or tuple or the nodes
+    from another graph).
+
+    >>> G.add_nodes_from(['a', 'b', 'c'])
+
+    **Edges:**
+
+    G can also be grown by adding edges.
+
+    Add one edge,
+
+    >>> G.add_edge('a', 'b')
+
+    a list of edges,
+
+    >>> G.add_edges_from([('a', 'b'), ('a', 'c')])
+
+    If some edges connect nodes not yet in the model, the nodes
+    are added automatically.  There are no errors when adding
+    nodes or edges that already exist.
+
+    **Shortcuts:**
+
+    Many common graph features allow python syntax for speed reporting.
+
+    >>> 'a' in G     # check if node in graph
+    True
+    >>> len(G)  # number of nodes in graph
+    3
+    """
+
+    def __init__(self, ebunch=None):
+        self.parent_node = None
+        self.children_nodes = []
+        super(NaiveBayesModel, self).__init__(ebunch)
```

Comment:
@ankurankan 
```__init__``` calls ```add_edges_from``` method which calls  ```add_edge``` method and I have ensured that the structure constraints are checked in ```add_edge``` method of ```NaiveBayes```.
```python
>>> from pgmpy.models import  NaiveBayesModel
>>> model = NaiveBayesModel([('a','b'), ('a','c'), ('b', 'c')])
Traceback (most recent call last):
  File "<pyshell#37>", line 1, in <module>
    model = NaiveBayesModel([('a','b'), ('a','c'), ('b', 'c')])
  File "D:\PC\t\naive_test.py", line 79, in __init__
    super(NaiveBayesModel, self).__init__(ebunch)
  File "C:\Python34\lib\site-packages\pgmpy-0.1.0-py3.4.egg\pgmpy\models\BayesianModel.py", line 85, in __init__
    self.add_edges_from(ebunch)
  File "C:\Python34\lib\site-packages\pgmpy-0.1.0-py3.4.egg\pgmpy\base\DirectedGraph.py", line 150, in add_edges_from
    self.add_edge(*edge, **kwargs)
  File "D:\PC\t\naive_test.py", line 103, in add_edge
    raise ValueError("Model can have only one parent node.")
ValueError: Model can have only one parent node.
```